### ISSUE TITLE
pcount returns the highest number

### ISSUE THREAD

**Issue by gesellkammer** [csound/csound#1070]
pcount returns the highest number of pargs passed to a given instrument instead of the actual number of pargs.

In the example below, I would expect "print ilen" to print 10, 14, 12 but it prints 10, then 14 and 14.

```csound

<CsoundSynthesizer>
<CsOptions>
</CsOptions>

<CsInstruments>
sr     = 44100
ksmps  = 64
nchnls = 2
0dbfs  = 1

instr 1
  ilen pcount
  print ilen
  turnoff
endin

</CsInstruments>

<CsScore>
i 1 0   1   4 5 6 7 8 9 10
i 1 0.1 1   4 5 6 7 8 9 10 11 12 13 14
i 1 0.2 1   4 5 6 7 8 9 10 11 12

; f0 3600

</CsScore>
</CsoundSynthesizer>
```

**Comment by vlazzarini** [comment]
I had a quick peek at this and all pcount does is report the current event pcnt, which is set at the score reading stage. I did not have time to trace it all back there, so I don't know where the bug is yet. Maybe @jpffitch will be quicker finding this.

**Comment by gesellkammer** [comment]
It has to do with "reusing" and instrument, since the old pargs of previous instances of the same instrument are still there. For example,

```csound

instr 1
  ilen pcount
  print ilen
  i0 = 1
  while i0 <= ilen do
    ival pindex i0
    print ival
    i0 = i0 + 1
  od
  turnoff
endin

<CsScore>
i 1 0   1   4 5 6 7 8 9 10
i 1 0.1 1   4 5 6 7 8 9 10 11 12 13 14
i 1 0.2 1   40 50 60 70 80 90 100 110 120

</CsScore>
```
Following the previous example, the third time instr 1 is scheduled, it prints the new args until 120 and then 13 and 14 (which were there from the second instance), meaning that the same event struct is being reused, the given pargs in the score are copied, but pcnt is not being reinitialized.

**Comment by vlazzarini** [comment]
yes, because there is only one currevent data that is used to hold pcnt and if the higher pfields are not used, they are not assigned to. I think  somewhere in score reading pcnt is retaining its peak value

**Comment by jpffitch** [comment]
Definitely the carry rules and not a bug.  I tried using -t0 and it  is clear

Forgotten how to switch carry off!

**Comment by gesellkammer** [comment]
Independently of carry being on/off, it would be much less confusing to have the opcode `pcount` return the number of explicit pfields used when scheduling.

**Comment by jpffitch** [comment]
The score expansion of carried values happens way before performance. 
How/where do you suggest this information is stored?

**Comment by gesellkammer** [comment]
You are right, pcount should in fact reflect carrying. Since the carry mechanism is well documented and can be turned off by either "!" or C 0 then anything else would be inconsistent. Sorry for the noise.
